daylay92/stacktrace-backend
# feature(notify): add email notification functionality

#### What does this PR do?
Enables users to get notified when their questions are answered.
#### Description of Task to be completed?
- Add a method to the Helpers class that performs the mailing operation.
- Modify the `create` method in the AnswerController class to accommodate the notification feature.
- Add a new abstraction method to the Answer services class to populate an Instance of an Answer.
- Write unit tests.
- Modify the README.md file.
#### How should this be manually tested?
- Clone the repo and cd into the project folder
- Checkout to `ft-notify-answers` branch and run `npm install`.
- Add a `.env` file to the root directory and include in it the following environment variables;
```
 MONGO_DEV_URI=your mongodb connection string for development
MONGO_URI=your mongodb connection string for the test environment
SECRET=whateveryoulike
SENDGRID_KEY=FHUIERH94389430303JFJEJLL.9494940-DJJ.09349030
EMAIL_TEMPLATE_ID=d-a1922b184048430088fd7d0qw816cd06
```
- Start the dev server by running `npm run start:dev`
- Using postman or insomnia, make a post request to `localhost:3000/api/v1/question/{questionId}/answer` to create an answer to a question
- Note that you would need to be logged in and to have a valid question Id to test out this feature.
- You can launch the api docs on your browser via `http://localhost:3000/api/v1/docs` for a head start.
- You can run `npm test` to see the results of the unit tests.
